### ISSUE TITLE
Make AggregationFuzzer reproducible

### ISSUE THREAD

**Issue by jayzhan211** [apache/datafusion#12903]
### Is your feature request related to a problem or challenge?

The PR [#12667](https://github.com/apache/datafusion/pull/12667) effectively generates tests to catch the bug. However, since it uses ThreadRng to generate data, the results are non-reproducible, making it difficult to pinpoint the root cause of the failure

### Describe the solution you'd like

I want the test to generate data randomly each time, but also have the option to pass a seed, so that failing test cases can be made reproducible

**Comment by Rachelint** [comment]
Yes, it is the one the most important features should be supported for me, too.
I am working on it.

**Comment by Rachelint** [comment]
take
